client/src/main/java/io/pravega/client/stream/ReaderGroup.java

```diff
@@ -177,6 +177,15 @@
      */
     ReaderSegmentDistribution getReaderSegmentDistribution();
 
+    /**
+     * The {@link StreamCut} provided to this API indicates that the stream has consumed data upto
+     * that point. Hence, it is safe for Pravega to truncate the stream using this {@link StreamCut} for this
+     * {@link ReaderGroup}.
+     *
+     * @param streamCuts Map of streams to their corresponding cuts with which truncation should be done.
+     */
+    void updateRetentionStreamCut(Map<Stream, StreamCut> streamCuts);
```

Comment:
Do we want to do all streams at once?

Reply:
i suppose this does not have to be "all the streams" that the RG is reading from but could well be a subset.

Reply:
My understanding is that this can either be a subset or all the streams that the RG is reading from.